core/src/main/java/io/parsingdata/metal/data/ParseResult.java

```diff
@@ -26,6 +29,28 @@ public ParseResult(final boolean succeeded, final Environment environment) {
         this.environment = environment;
     }
 
+    public void handleCallbacks(final Token token) {
+        handleCallbacks(environment.callbacks, token);
+    }
+
+    private void handleCallbacks(final TokenCallbackList callbacks, final Token token) {
```

Comment:
Excellent point! I had a hard time coming up with a good test however, which indicated that it probably shouldn't be a public API. To fix this I moved the method to the `Token` base class, which greatly simplified the implementation.